### ISSUE TITLE
integration.modules.test_mine.MineTest.test_mine_delete

### ISSUE THREAD

**Issue by dwoz** [saltstack/salt#52087]
```
integration.modules.test_mine.MineTest.test_mine_delete
--------------------------------------------------------------------------------
2018.3 failed salt-fedora-28-py2
--------------------------------------------------------------------------------
--------------------------------------------------------------------------------
2019.2 failed salt-fedora-28-py2
--------------------------------------------------------------------------------
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
{u'kernelrelease': u'4.20.11-100.fc28.x86_64', u'selinux': {u'enforced': u'Enforcing', u'enabled': True}, u'ip4_interfaces': {u'lo': [u'127.0.0.1'], u'eth0': [u'10.200.11.193']}, u'saltversioninfo': [2019, 2, 0, 0], u'systemd': {u'version': u'238', u'features': u'+PAM +AUDIT +SELINUX +IMA -APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD +IDN2 -IDN +PCRE2 default-hierarchy=hybrid'}, u'uuid': u'ec2cfaae-ccd4-4776-4fd0-950a9a96c42c', u'hwaddr_interfaces': {u'lo': u'00:00:00:00:00:00', u'eth0': u'06:0f:6f:64:a1:1a'}, u'init': u'systemd', u'virtual_subtype': u'Xen HVM DomU', u'cpu_flags': [u'fpu', u'vme', u'de', u'pse', u'tsc', u'msr', u'pae', u'mce', u'cx8', u'apic', u'sep', u'mtrr', u'pge', u'mca', u'cmov', u'pat', u'pse36', u'clflush', u'mmx', u'fxsr', u'sse', u'sse2', u'ht', u'syscall', u'nx', u'pdpe1gb', u'rdtscp', u'lm', u'constant_tsc', u'rep_good', u'nopl', u'xtopology', u'cpuid', u'pni', u'pclmulqdq', u'ssse3', u'fma', u'cx16', u'pcid', u'sse4_1', u'sse4_2', u'x2apic', u'movbe', u'popcnt', u'tsc_deadline_timer', u'aes', u'xsave', u'avx', u'f16c', u'rdrand', u'hypervisor', u'lahf_lm', u'abm', u'cpuid_fault', u'invpcid_single', u'pti', u'fsgsbase', u'bmi1', u'avx2', u'smep', u'bmi2', u'erms', u'invpcid', u'xsaveopt'], u'match': u'maker', u'cpu_model': u'Intel(R) Xeon(R) CPU E5-2686 v4 @ 2.30GHz', u'oscodename': u'Twenty Eight', u'virtual_hv_features': u'00000705', u'gpus': [], u'virtual_hv_version': u'4.2.amazon', u'num_gpus': 0, u'disks': [], u'productname': u'HVM domU', u'osarch': u'x86_64', u'machine_id': u'5d8e070d7b6746f99ccd1dc54804bb43', u'biosversion': u'4.2.amazon', u'kernel': u'Linux', u'domain': u'us-west-2.compute.internal', u'zookeeper': {u'username': u'daniel', u'password': u'test', u'hosts': u'localhost:2181', u'default_acl': [{u'username': u'daniel', u'read': True, u'create': True, u'write': True, u'admin': True, u'password': u'test', u'delete': True}], u'prod': {u'username': u'daniel', u'password': u'test', u'hosts': u'localhost:2181', u'default_acl': [{u'username': u'daniel', u'read': True, u'create': True, u'write': True, u'admin': True, u'password': u'test', u'delete': True}]}}, u'tokenv2': {u'keystone.endpoint': u'http://localhost:35357/v2.0', u'keystone.token': u'administrator'}, u'tokenv3': {u'keystone.endpoint': u'http://localhost:35357/v3', u'keystone.token': u'administrator'}, u'ip_interfaces': {u'lo': [u'127.0.0.1', u'::1'], u'eth0': [u'10.200.11.193', u'fe80::40f:6fff:fe64:a11a']}, u'mem_total': 16036, u'mdadm': [], u'nodename': u'ip-10-200-11-193.us-west-2.compute.internal', u'companions': {u'one': [u'susan', u'ian', u'barbara']}, u'osfullname': u'Fedora', u'test_grain': u'cheese', u'localhost': u'ip-10-200-11-193.us-west-2.compute.internal', u'demov3': {u'keystone.password': u'demopass', u'keystone.tenant': u'demo', u'keystone.user': u'demo', u'keystone.auth_url': u'http://127.0.0.1:5000/v3/'}, u'demov2': {u'keystone.password': u'demopass', u'keystone.tenant': u'demo', u'keystone.user': u'demo', u'keystone.auth_url': u'http://127.0.0.1:5000/v2.0/'}, u'lsb_distrib_release': u'28', u'salttesting-grain-key': [], u'saltpath': u'/tmp/kitchen/testing/salt', u'biosreleasedate': u'08/24/2006', u'host': u'ip-10-200-11-193', u'osfinger': u'Fedora-28', u'manufacturer': u'Xen', u'kernelversion': u'#1 SMP Wed Feb 20 15:51:24 UTC 2019', u'ip4_gw': u'10.200.0.1', u'virtual_hv_version_info': [u'4', u'2', u'.amazon'], u'fqdn': u'ip-10-200-11-193.us-west-2.compute.internal', u'alot': u'many', u'os': u'Fedora', u'pythonpath': [u'/tmp/kitchen/testing', u'/tmp/kitchen/testing/tests', u'/usr/lib/python2.7/site-packages/git/ext/gitdb', u'/tmp/salt-tests-tmpdir/scripts', u'/usr/lib/python27.zip', u'/usr/lib64/python2.7', u'/usr/lib64/python2.7/plat-linux2', u'/usr/lib64/python2.7/lib-tk', u'/usr/lib64/python2.7/lib-old', u'/usr/lib64/python2.7/lib-dynload', u'/usr/lib64/python2.7/site-packages', u'/usr/lib/python2.7/site-packages', u'/usr/lib/python2.7/site-packages/gitdb/ext/smmap'], u'a_custom': {u'k2': u'v2'}, u'path': u'/tmp/kitchen/testing/tests/integration/mockbin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin', u'SSDs': [u'xvda'], u'ps': u'ps -efHww', u'locale_info': {u'timezone': u'UTC', u'detectedencoding': u'UTF-8', u'defaultlanguage': u'en_US', u'defaultencoding': u'UTF-8'}, u'ip_gw': True, u'script': u'grail', u'osrelease_info': [28], u'gid': 0, u'master': u'localhost', u'ipv4': [u'10.200.11.193', u'127.0.0.1'], u'dns': {u'domain': u'', u'sortlist': [], u'nameservers': [u'10.200.0.2'], u'ip4_nameservers': [u'10.200.0.2'], u'search': [u'us-west-2.compute.internal'], u'ip6_nameservers': [], u'options': []}, u'ipv6': [u'::1', u'fe80::40f:6fff:fe64:a11a'], u'server_id': 266635400, u'lsb_distrib_id': u'Fedora', u'username': u'root', u'shell': u'/bin/bash', u'saltversion': u'2019.2.0-252-g01bcff0', u'fqdns': [], u'zfs_feature_flags': False, u'groupname': u'root', u'cpuarch': u'x86_64', u'virtual_hv_features_list': [u'writable_page_tables', u'auto_translated_physmap', u'hvm_callback_vector', u'hvm_safe_pvclock', u'hvm_pirqs'], u'uid': 0, u'zmqversion': u'4.1.6', u'serialnumber': u'ec2cfaae-ccd4-4776-4fd0-950a9a96c42c', u'pid': 8340, u'level1': {u'level2': u'foo'}, u'planets': [u'mercury', u'venus', u'earth', u'mars'], u'zfs_support': False, u'id': u'minion', u'osrelease': u'28', u'adminv2': {u'keystone.password': u'adminpass', u'keystone.tenant': u'admin', u'keystone.user': u'admin', u'keystone.auth_url': u'http://127.0.0.1:5000/v2.0/'}, u'adminv3': {u'keystone.password': u'adminpass', u'keystone.tenant': u'admin', u'keystone.user': u'admin', u'keystone.auth_url': u'http://127.0.0.1:5000/v3/'}, u'ip6_interfaces': {u'lo': [u'::1'], u'eth0': [u'fe80::40f:6fff:fe64:a11a']}, u'num_cpus': 4, u'virtual': u'xen', u'fqdn_ip4': [u'10.200.11.193'], u'fqdn_ip6': [u'fe80::40f:6fff:fe64:a11a'], u'custom_grain_test': u'itworked', u'ip6_gw': False, u'osmajorrelease': 28, u'os_family': u'RedHat', u'pythonversion': [2, 7, 15, u'final', 0], u'pythonexecutable': u'/usr/bin/python2.7', u'swap_total': 2047, u'lsb_distrib_codename': u'', u'setgrain': u'grainval'} != None
................................................................................
Traceback (most recent call last):
  File "/tmp/kitchen/testing/tests/integration/modules/test_mine.py", line 140, in test_mine_delete
    self.assertEqual(ret_grains_deleted.get('minion', None), None)
AssertionError: {u'kernelrelease': u'4.20.11-100.fc28.x86_64', u'selinux': {u'enforced': u'Enforcing', u'enabled': True}, u'ip4_interfaces': {u'lo': [u'127.0.0.1'], u'eth0': [u'10.200.11.193']}, u'saltversioninfo': [2019, 2, 0, 0], u'systemd': {u'version': u'238', u'features': u'+PAM +AUDIT +SELINUX +IMA -APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD +IDN2 -IDN +PCRE2 default-hierarchy=hybrid'}, u'uuid': u'ec2cfaae-ccd4-4776-4fd0-950a9a96c42c', u'hwaddr_interfaces': {u'lo': u'00:00:00:00:00:00', u'eth0': u'06:0f:6f:64:a1:1a'}, u'init': u'systemd', u'virtual_subtype': u'Xen HVM DomU', u'cpu_flags': [u'fpu', u'vme', u'de', u'pse', u'tsc', u'msr', u'pae', u'mce', u'cx8', u'apic', u'sep', u'mtrr', u'pge', u'mca', u'cmov', u'pat', u'pse36', u'clflush', u'mmx', u'fxsr', u'sse', u'sse2', u'ht', u'syscall', u'nx', u'pdpe1gb', u'rdtscp', u'lm', u'constant_tsc', u'rep_good', u'nopl', u'xtopology', u'cpuid', u'pni', u'pclmulqdq', u'ssse3', u'fma', u'cx16', u'pcid', u'sse4_1', u'sse4_2', u'x2apic', u'movbe', u'popcnt', u'tsc_deadline_timer', u'aes', u'xsave', u'avx', u'f16c', u'rdrand', u'hypervisor', u'lahf_lm', u'abm', u'cpuid_fault', u'invpcid_single', u'pti', u'fsgsbase', u'bmi1', u'avx2', u'smep', u'bmi2', u'erms', u'invpcid', u'xsaveopt'], u'match': u'maker', u'cpu_model': u'Intel(R) Xeon(R) CPU E5-2686 v4 @ 2.30GHz', u'oscodename': u'Twenty Eight', u'virtual_hv_features': u'00000705', u'gpus': [], u'virtual_hv_version': u'4.2.amazon', u'num_gpus': 0, u'disks': [], u'productname': u'HVM domU', u'osarch': u'x86_64', u'machine_id': u'5d8e070d7b6746f99ccd1dc54804bb43', u'biosversion': u'4.2.amazon', u'kernel': u'Linux', u'domain': u'us-west-2.compute.internal', u'zookeeper': {u'username': u'daniel', u'password': u'test', u'hosts': u'localhost:2181', u'default_acl': [{u'username': u'daniel', u'read': True, u'create': True, u'write': True, u'admin': True, u'password': u'test', u'delete': True}], u'prod': {u'username': u'daniel', u'password': u'test', u'hosts': u'localhost:2181', u'default_acl': [{u'username': u'daniel', u'read': True, u'create': True, u'write': True, u'admin': True, u'password': u'test', u'delete': True}]}}, u'tokenv2': {u'keystone.endpoint': u'http://localhost:35357/v2.0', u'keystone.token': u'administrator'}, u'tokenv3': {u'keystone.endpoint': u'http://localhost:35357/v3', u'keystone.token': u'administrator'}, u'ip_interfaces': {u'lo': [u'127.0.0.1', u'::1'], u'eth0': [u'10.200.11.193', u'fe80::40f:6fff:fe64:a11a']}, u'mem_total': 16036, u'mdadm': [], u'nodename': u'ip-10-200-11-193.us-west-2.compute.internal', u'companions': {u'one': [u'susan', u'ian', u'barbara']}, u'osfullname': u'Fedora', u'test_grain': u'cheese', u'localhost': u'ip-10-200-11-193.us-west-2.compute.internal', u'demov3': {u'keystone.password': u'demopass', u'keystone.tenant': u'demo', u'keystone.user': u'demo', u'keystone.auth_url': u'http://127.0.0.1:5000/v3/'}, u'demov2': {u'keystone.password': u'demopass', u'keystone.tenant': u'demo', u'keystone.user': u'demo', u'keystone.auth_url': u'http://127.0.0.1:5000/v2.0/'}, u'lsb_distrib_release': u'28', u'salttesting-grain-key': [], u'saltpath': u'/tmp/kitchen/testing/salt', u'biosreleasedate': u'08/24/2006', u'host': u'ip-10-200-11-193', u'osfinger': u'Fedora-28', u'manufacturer': u'Xen', u'kernelversion': u'#1 SMP Wed Feb 20 15:51:24 UTC 2019', u'ip4_gw': u'10.200.0.1', u'virtual_hv_version_info': [u'4', u'2', u'.amazon'], u'fqdn': u'ip-10-200-11-193.us-west-2.compute.internal', u'alot': u'many', u'os': u'Fedora', u'pythonpath': [u'/tmp/kitchen/testing', u'/tmp/kitchen/testing/tests', u'/usr/lib/python2.7/site-packages/git/ext/gitdb', u'/tmp/salt-tests-tmpdir/scripts', u'/usr/lib/python27.zip', u'/usr/lib64/python2.7', u'/usr/lib64/python2.7/plat-linux2', u'/usr/lib64/python2.7/lib-tk', u'/usr/lib64/python2.7/lib-old', u'/usr/lib64/python2.7/lib-dynload', u'/usr/lib64/python2.7/site-packages', u'/usr/lib/python2.7/site-packages', u'/usr/lib/python2.7/site-packages/gitdb/ext/smmap'], u'a_custom': {u'k2': u'v2'}, u'path': u'/tmp/kitchen/testing/tests/integration/mockbin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin', u'SSDs': [u'xvda'], u'ps': u'ps -efHww', u'locale_info': {u'timezone': u'UTC', u'detectedencoding': u'UTF-8', u'defaultlanguage': u'en_US', u'defaultencoding': u'UTF-8'}, u'ip_gw': True, u'script': u'grail', u'osrelease_info': [28], u'gid': 0, u'master': u'localhost', u'ipv4': [u'10.200.11.193', u'127.0.0.1'], u'dns': {u'domain': u'', u'sortlist': [], u'nameservers': [u'10.200.0.2'], u'ip4_nameservers': [u'10.200.0.2'], u'search': [u'us-west-2.compute.internal'], u'ip6_nameservers': [], u'options': []}, u'ipv6': [u'::1', u'fe80::40f:6fff:fe64:a11a'], u'server_id': 266635400, u'lsb_distrib_id': u'Fedora', u'username': u'root', u'shell': u'/bin/bash', u'saltversion': u'2019.2.0-252-g01bcff0', u'fqdns': [], u'zfs_feature_flags': False, u'groupname': u'root', u'cpuarch': u'x86_64', u'virtual_hv_features_list': [u'writable_page_tables', u'auto_translated_physmap', u'hvm_callback_vector', u'hvm_safe_pvclock', u'hvm_pirqs'], u'uid': 0, u'zmqversion': u'4.1.6', u'serialnumber': u'ec2cfaae-ccd4-4776-4fd0-950a9a96c42c', u'pid': 8340, u'level1': {u'level2': u'foo'}, u'planets': [u'mercury', u'venus', u'earth', u'mars'], u'zfs_support': False, u'id': u'minion', u'osrelease': u'28', u'adminv2': {u'keystone.password': u'adminpass', u'keystone.tenant': u'admin', u'keystone.user': u'admin', u'keystone.auth_url': u'http://127.0.0.1:5000/v2.0/'}, u'adminv3': {u'keystone.password': u'adminpass', u'keystone.tenant': u'admin', u'keystone.user': u'admin', u'keystone.auth_url': u'http://127.0.0.1:5000/v3/'}, u'ip6_interfaces': {u'lo': [u'::1'], u'eth0': [u'fe80::40f:6fff:fe64:a11a']}, u'num_cpus': 4, u'virtual': u'xen', u'fqdn_ip4': [u'10.200.11.193'], u'fqdn_ip6': [u'fe80::40f:6fff:fe64:a11a'], u'custom_grain_test': u'itworked', u'ip6_gw': False, u'osmajorrelease': 28, u'os_family': u'RedHat', u'pythonversion': [2, 7, 15, u'final', 0], u'pythonexecutable': u'/usr/bin/python2.7', u'swap_total': 2047, u'lsb_distrib_codename': u'', u'setgrain': u'grainval'} != None
```

**Comment by waynew** [comment]
I cannot reproduce this, either :-\

**Comment by waynew** [comment]
Bah! I did! I'm about 100% sure this test straddled the `mine_interval`, so the thing it was supposed to delete got stuck back into the mine. Or *something* got it back into the mine, somehow.

I was able to reproduce (finally) by adding a `time.sleep` after the delete and before the get. Then while the tests are running:

```
salt -c /tmp/salt-tests-tmpdir/config \* mine.send grains.items
```
Then? Kaboom. Test fails.

Got an incoming PR to hopefully make this test less flaky
